sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringOperations.scala

```diff
@@ -501,6 +501,32 @@ case class StringFormat(children: Expression*) extends Expression with CodegenFa
     }
   }
 
+  override def genCode(ctx: CodeGenContext, ev: GeneratedExpressionCode): String = {
+    val pattern = children.head.gen(ctx)
+
+    val argListGen = children.tail.map(_.gen(ctx))
+    val argListCode = argListGen.map(_.code + "\n")
+    val argListString = argListGen.foldLeft("")((s, v) => s + s", ${v.primitive}")
```

Comment:
`s",  ${v.isNull} ? null : ${v.primitive}"` Doesn't compile because of: `Incompatible expression types "void" and "int"`

Casting the null to the Boxed type, throws a null pointer exception: 
```Java
int primitive6 = 0;
Object o = (true) ? (Integer) null : primitive6;
```